zerver/webhooks/airbyte/view.py

```diff
@@ -0,0 +1,95 @@
+* **Records:** {records_emitted} emitted, {records_committed} committed
+* **Bytes:** {bytes_emitted_formatted} emitted, {bytes_committed_formatted} committed
+* **Duration:** {duration_formatted}
+"""
+
+
+def extract_data_from_payload(payload_data: WildValue) -> dict[str, str | int | bool]:
+    return {
+        "workspace_name": payload_data["workspace"]["name"].tame(check_string),
+        "connection_name": payload_data["connection"]["name"].tame(check_string),
+        "source_name": payload_data["source"]["name"].tame(check_string),
+        "destination_name": payload_data["destination"]["name"].tame(check_string),
+        "connection_url": payload_data["connection"]["url"].tame(check_string),
+        "source_url": payload_data["source"]["url"].tame(check_string),
+        "destination_url": payload_data["destination"]["url"].tame(check_string),
+        "successful_sync": payload_data["success"].tame(check_bool),
+        "duration_formatted": payload_data["durationFormatted"].tame(check_string),
+        "records_emitted": payload_data["recordsEmitted"].tame(check_int),
+        "records_committed": payload_data["recordsCommitted"].tame(check_int),
+        "bytes_emitted_formatted": payload_data["bytesEmittedFormatted"].tame(check_string),
+        "bytes_committed_formatted": payload_data["bytesCommittedFormatted"].tame(check_string),
+        "error_message": payload_data.get("errorMessage").tame(check_string)
+        if payload_data.get("errorMessage")
+        else "No error",
+    }
+
+
+def format_message_from_data(data: dict[str, str | int | bool]) -> str:
+    content = AIRBYTE_MESSAGE_TEMPLATE.format(
+        sync_status_emoji=":green_circle:" if data["successful_sync"] else ":red_circle:",
+        status="succeeded" if data["successful_sync"] else "failed",
+        connection_name=data["connection_name"],
+        connection_url=data["connection_url"],
+        source_name=data["source_name"],
+        source_url=data["source_url"],
+        destination_name=data["destination_name"],
+        destination_url=data["destination_url"],
+        duration_formatted=data["duration_formatted"],
+        records_emitted=data["records_emitted"],
+        records_committed=data["records_committed"],
+        bytes_emitted_formatted=data["bytes_emitted_formatted"],
+        bytes_committed_formatted=data["bytes_committed_formatted"],
+    )
+
+    if not data["successful_sync"]:
+        error_message = data["error_message"]
+        content += f"\n**Error message:** {error_message}"
+
+    return content
+
+
+def create_topic_from_data(data: dict[str, str | int | bool]) -> str:
+    return AIRBYTE_TOPIC_TEMPLATE.format(
+        workspace=data["workspace_name"],
+        connection=data["connection_name"],
+    )
+
+
+def create_airbyte_message(payload: WildValue) -> tuple[str, str]:
+    data = extract_data_from_payload(payload["data"])
+    content = format_message_from_data(data)
+    topic = create_topic_from_data(data)
```

Comment:
```suggestion
    if "data" in payload:
        data = extract_data_from_payload(payload["data"])
        content = format_message_from_data(data)
        topic = create_topic_from_data(data)
    else:
        content = payload["text"].tame(check_string)
        topic = "Airbyte"
```

Reply:
This is the simplest implementation I'd be comfortable with personally.

Reply:
Hmm. This looks good.

Reply:
> Maybe the fallback topic should be "Airbyte notification" instead of just "Airbyte"?

Agree, @laurynmm.

I actually had `topic = "Airbyte Notification"` originally, but then checked a few random integration implementations and saw that Dropbox had `topic_name = "Dropbox"`. That's the only reason I changed it.